### PR TITLE
Remove explicit stream finish

### PR DESCRIPTION
**Pull request by lijunwangs**
#### Problem

stream finish is redundant as StreamDrop is already doing that.
Removing that will allow next call to be made faster without waiting for the stream to be finished.

#### Summary of Changes

removed explicit stream finish

Fixes #

### CODE REVIEW

**Comment by lijunwangs** [comment]
Tests:

Remove finish


[2024-08-27T03:19:06.899749826Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |     119661.60 | 337158 
[2024-08-27T03:19:06.899756703Z INFO  solana_bench_tps::bench] 
    Average max TPS: 119661.60, 0 nodes had 0 TPS
[2024-08-27T03:19:06.899759446Z INFO  solana_bench_tps::bench] 
    Highest TPS: 119661.60 sampling period 1s max transactions: 337158 clients: 1 drop rate: 0.81
[2024-08-27T03:19:06.899762153Z INFO  solana_bench_tps::bench] 	Average TPS: 11026.231


[2024-08-27T03:20:13.052122723Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      67628.32 | 360135 
[2024-08-27T03:20:13.052126581Z INFO  solana_bench_tps::bench] 
    Average max TPS: 67628.32, 0 nodes had 0 TPS
[2024-08-27T03:20:13.052129360Z INFO  solana_bench_tps::bench] 
    Highest TPS: 67628.32 sampling period 1s max transactions: 360135 clients: 1 drop rate: 0.81
[2024-08-27T03:20:13.052131933Z INFO  solana_bench_tps::bench] 	Average TPS: 11814.04


[2024-08-27T03:21:27.073280575Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      42699.27 | 298108 
[2024-08-27T03:21:27.073290647Z INFO  solana_bench_tps::bench] 
    Average max TPS: 42699.27, 0 nodes had 0 TPS
[2024-08-27T03:21:27.073297927Z INFO  solana_bench_tps::bench] 
    Highest TPS: 42699.27 sampling period 1s max transactions: 298108 clients: 1 drop rate: 0.83
[2024-08-27T03:21:27.073301372Z INFO  solana_bench_tps::bench] 	Average TPS: 9650.714

[2024-08-27T04:01:30.855469155Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      57559.41 | 301227 
[2024-08-27T04:01:30.855473317Z INFO  solana_bench_tps::bench] 
    Average max TPS: 57559.41, 0 nodes had 0 TPS
[2024-08-27T04:01:30.855476117Z INFO  solana_bench_tps::bench] 
    Highest TPS: 57559.41 sampling period 1s max transactions: 301227 clients: 1 drop rate: 0.85
[2024-08-27T04:01:30.855478908Z INFO  solana_bench_tps::bench] 	Average TPS: 9858.805


[2024-08-27T04:10:58.127741634Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      44547.83 | 239731 
[2024-08-27T04:10:58.127745678Z INFO  solana_bench_tps::bench] 
    Average max TPS: 44547.83, 0 nodes had 0 TPS
[2024-08-27T04:10:58.127748489Z INFO  solana_bench_tps::bench] 
    Highest TPS: 44547.83 sampling period 1s max transactions: 239731 clients: 1 drop rate: 0.85
[2024-08-27T04:10:58.127751265Z INFO  solana_bench_tps::bench] 	Average TPS: 7676.3643


[2024-08-27T04:12:53.375344215Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      37728.53 | 225638 
[2024-08-27T04:12:53.375353453Z INFO  solana_bench_tps::bench] 
    Average max TPS: 37728.53, 0 nodes had 0 TPS
[2024-08-27T04:12:53.375356111Z INFO  solana_bench_tps::bench] 
    Highest TPS: 37728.53 sampling period 1s max transactions: 225638 clients: 1 drop rate: 0.87
[2024-08-27T04:12:53.375361379Z INFO  solana_bench_tps::bench] 	Average TPS: 7467.27



[2024-08-27T04:16:53.013411655Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      34871.02 | 230517 
[2024-08-27T04:16:53.013415730Z INFO  solana_bench_tps::bench] 
    Average max TPS: 34871.02, 0 nodes had 0 TPS
[2024-08-27T04:16:53.013418782Z INFO  solana_bench_tps::bench] 
    Highest TPS: 34871.02 sampling period 1s max transactions: 230517 clients: 1 drop rate: 0.87
[2024-08-27T04:16:53.013421537Z INFO  solana_bench_tps::bench] 	Average TPS: 7332.2646


[2024-08-27T04:22:00.525223852Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      26258.62 | 288714 
[2024-08-27T04:22:00.525231491Z INFO  solana_bench_tps::bench] 
    Average max TPS: 26258.62, 0 nodes had 0 TPS
[2024-08-27T04:22:00.525235187Z INFO  solana_bench_tps::bench] 
    Highest TPS: 26258.62 sampling period 1s max transactions: 288714 clients: 1 drop rate: 0.78
[2024-08-27T04:22:00.525238231Z INFO  solana_bench_tps::bench] 	Average TPS: 9460.518


[2024-08-27T04:23:33.494167013Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |     208339.42 | 333608 
[2024-08-27T04:23:33.494173363Z INFO  solana_bench_tps::bench] 
    Average max TPS: 208339.42, 0 nodes had 0 TPS
[2024-08-27T04:23:33.494176756Z INFO  solana_bench_tps::bench] 
    Highest TPS: 208339.42 sampling period 1s max transactions: 333608 clients: 1 drop rate: 0.73
[2024-08-27T04:23:33.494181093Z INFO  solana_bench_tps::bench] 	Average TPS: 10918.707


[2024-08-27T04:26:46.127309907Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      40996.89 | 209066 
[2024-08-27T04:26:46.127315620Z INFO  solana_bench_tps::bench] 
    Average max TPS: 40996.89, 0 nodes had 0 TPS
[2024-08-27T04:26:46.127319132Z INFO  solana_bench_tps::bench] 
    Highest TPS: 40996.89 sampling period 1s max transactions: 209066 clients: 1 drop rate: 0.89
[2024-08-27T04:26:46.127323390Z INFO  solana_bench_tps::bench] 	Average TPS: 6519.779


[2024-08-27T04:30:03.535314829Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      40485.67 | 193588 
[2024-08-27T04:30:03.535318892Z INFO  solana_bench_tps::bench] 
    Average max TPS: 40485.67, 0 nodes had 0 TPS
[2024-08-27T04:30:03.535321369Z INFO  solana_bench_tps::bench] 
    Highest TPS: 40485.67 sampling period 1s max transactions: 193588 clients: 1 drop rate: 0.89
[2024-08-27T04:30:03.535324414Z INFO  solana_bench_tps::bench] 	Average TPS: 6378.0034





With finish:


[2024-08-27T04:35:42.854548034Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      34401.23 | 251795 
[2024-08-27T04:35:42.854553747Z INFO  solana_bench_tps::bench] 
    Average max TPS: 34401.23, 0 nodes had 0 TPS
[2024-08-27T04:35:42.854564108Z INFO  solana_bench_tps::bench] 
    Highest TPS: 34401.23 sampling period 1s max transactions: 251795 clients: 1 drop rate: 0.84
[2024-08-27T04:35:42.854575197Z INFO  solana_bench_tps::bench] 	Average TPS: 8372.723
[2024-08-27T04:35:42.854554311Z INFO  solana_metrics::metrics] datapoint: bench-tps-lamport_balance balance=28880960150i



[2024-08-27T04:39:36.154322986Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      55594.46 | 217749 
[2024-08-27T04:39:36.154326808Z INFO  solana_bench_tps::bench] 
    Average max TPS: 55594.46, 0 nodes had 0 TPS
[2024-08-27T04:39:36.154329631Z INFO  solana_bench_tps::bench] 
    Highest TPS: 55594.46 sampling period 1s max transactions: 217749 clients: 1 drop rate: 0.87
[2024-08-27T04:39:36.154340684Z INFO  solana_bench_tps::bench] 	Average TPS: 6658.179

[2024-08-27T05:58:37.383945958Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      38718.09 | 220350 
[2024-08-27T05:58:37.383950083Z INFO  solana_bench_tps::bench] 
    Average max TPS: 38718.09, 0 nodes had 0 TPS
[2024-08-27T05:58:37.383953013Z INFO  solana_bench_tps::bench] 
    Highest TPS: 38718.09 sampling period 1s max transactions: 220350 clients: 1 drop rate: 0.87
[2024-08-27T05:58:37.383955917Z INFO  solana_bench_tps::bench] 	Average TPS: 6959.633


[2024-08-27T06:01:55.251171119Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      64582.13 | 397494 
[2024-08-27T06:01:55.251175934Z INFO  solana_bench_tps::bench] 
    Average max TPS: 64582.13, 0 nodes had 0 TPS
[2024-08-27T06:01:55.251178296Z INFO  solana_bench_tps::bench] 
    Highest TPS: 64582.13 sampling period 1s max transactions: 397494 clients: 1 drop rate: 0.75
[2024-08-27T06:01:55.251181581Z INFO  solana_bench_tps::bench] 	Average TPS: 12542.041


[2024-08-27T06:04:54.200298394Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      59009.90 | 221467 
[2024-08-27T06:04:54.200302145Z INFO  solana_bench_tps::bench] 
    Average max TPS: 59009.90, 0 nodes had 0 TPS
[2024-08-27T06:04:54.200304434Z INFO  solana_bench_tps::bench] 
    Highest TPS: 59009.90 sampling period 1s max transactions: 221467 clients: 1 drop rate: 0.87
[2024-08-27T06:04:54.200307409Z INFO  solana_bench_tps::bench] 	Average TPS: 7251.6035


[2024-08-27T16:19:48.770250840Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      47243.82 | 279543 
[2024-08-27T16:19:48.770259809Z INFO  solana_bench_tps::bench] 
    Average max TPS: 47243.82, 0 nodes had 0 TPS
[2024-08-27T16:19:48.770263077Z INFO  solana_bench_tps::bench] 
    Highest TPS: 47243.82 sampling period 1s max transactions: 279543 clients: 1 drop rate: 0.79
[2024-08-27T16:19:48.770265774Z INFO  solana_bench_tps::bench] 	Average TPS: 9295.732

[2024-08-27T16:45:27.853456505Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      29285.67 | 56799 
[2024-08-27T16:45:27.853460461Z INFO  solana_bench_tps::bench] 
    Average max TPS: 29285.67, 0 nodes had 0 TPS
[2024-08-27T16:45:27.853462971Z INFO  solana_bench_tps::bench] 
    Highest TPS: 29285.67 sampling period 1s max transactions: 56799 clients: 1 drop rate: 0.97
[2024-08-27T16:45:27.853465971Z INFO  solana_bench_tps::bench] 	Average TPS: 1833.7365

[2024-08-27T17:03:15.795158258Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      88490.29 | 207106 
[2024-08-27T17:03:15.795163317Z INFO  solana_bench_tps::bench] 
    Average max TPS: 88490.29, 0 nodes had 0 TPS
[2024-08-27T17:03:15.795167160Z INFO  solana_bench_tps::bench] 
    Highest TPS: 88490.29 sampling period 1s max transactions: 207106 clients: 1 drop rate: 0.85
[2024-08-27T17:03:15.795171571Z INFO  solana_bench_tps::bench] 	Average TPS: 6402.1084



[2024-08-27T17:07:54.294352195Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      80054.30 | 341621 
[2024-08-27T17:07:54.294357293Z INFO  solana_bench_tps::bench] 
    Average max TPS: 80054.30, 0 nodes had 0 TPS
[2024-08-27T17:07:54.294359703Z INFO  solana_bench_tps::bench] 
    Highest TPS: 80054.30 sampling period 1s max transactions: 341621 clients: 1 drop rate: 0.76
[2024-08-27T17:07:54.294362563Z INFO  solana_bench_tps::bench] 	Average TPS: 10854.25
[2024-08-27T17:07:54.294371976Z INFO  solana_metrics::metrics] datapoint: bench-tps-lamport_balance balance=30964440150i


[2024-08-27T17:15:04.633871152Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      40748.52 | 313114 
[2024-08-27T17:15:04.633875246Z INFO  solana_bench_tps::bench] 
    Average max TPS: 40748.52, 0 nodes had 0 TPS
[2024-08-27T17:15:04.633877674Z INFO  solana_bench_tps::bench] 
    Highest TPS: 40748.52 sampling period 1s max transactions: 313114 clients: 1 drop rate: 0.79
[2024-08-27T17:15:04.633880858Z INFO  solana_bench_tps::bench] 	Average TPS: 10349.3


[2024-08-27T17:37:15.418530398Z INFO  solana_bench_tps::bench] http://35.233.177.221:8899 |      31692.88 | 195333 
[2024-08-27T17:37:15.418535759Z INFO  solana_bench_tps::bench] 
    Average max TPS: 31692.88, 0 nodes had 0 TPS
[2024-08-27T17:37:15.418539641Z INFO  solana_bench_tps::bench] 
    Highest TPS: 31692.88 sampling period 1s max transactions: 195333 clients: 1 drop rate: 0.88
[2024-08-27T17:37:15.418543406Z INFO  solana_bench_tps::bench] 	Average TPS: 6219.16


In summary with bench-tps,

without finish on average: 8918

With finish: on average 7884

The different is not as big as in the case of serially sending the transactions in a single thread -- as concurrent streams send currently can reduce the latency impact.

**Comment by KirillLykov** [comment]
What is RTT for the client/server in these tests? I'm asking because from my observations the higher is RTT, the higher is impact because calling `finish` makes it to wait for ACK, so it is like having RW=1 transaction (my understanding, might be wrong in the interpretation).

Did you test it with multistream without finish or with "one" stream?

**Review by alessandrod — APPROVED** [review]

**Comment by lijunwangs** [comment]
> What is RTT for the client/server in these tests? I'm asking because from my observations the higher is RTT, the higher is impact because calling `finish` makes it to wait for ACK, so it is like having RW=1 transaction (my understanding, might be wrong in the interpretation).
> 
> Did you test it with multistream without finish or with "one" stream?

the bench-tps is multi stream. The result is from local GCE cluster with low RTT < 1MS.
